### PR TITLE
fix: do not close sidekick if loaded externally

### LINKED ISSUE #478: Sidekick loaded via messaging API gets closed again after tab switch

### PR DESCRIPTION
**Pull request by rofe**
Fix #478

### CODE REVIEW

**Review by kptdobe — APPROVED** [review]
Looks good but is it enough... ? does it open and close the sidekick if I click the extension button ? What happens if I close the sidekick with the close button, how do I reopen it ?

**Comment by rofe** [comment]
You,re right, it's more complicated...

**Comment by rofe** [comment]
@kptdobe more refactoring was needed to enable all scenarios:
- extended `DISCOVERY_CACHE` to a new cache holding both dynamic (discovered) and static (added by the page) entries
- dynamic entries expire after 2 hours (same ad before)
- the cache is persisted in session storage now, so all entries expire at the end of the browser session (previously, it was the end of the service worker lifetime, which can be significantly shorter)

File: src/extension/utils.js [hunk 1]
```diff
@@ -326,8 +362,20 @@ export function getProjectMatches(configs, tabUrl) {
...
+      if (e.owner === owner && e.repo === repo) {
+        // avoid duplicates
+        if (!matches.find((m) => m.owner === owner && m.repo === repo)) {
+          return true;
+        }
+      }
+      return false;
+    });
+    matches = matches.concat(filteredByUrlCache);
   });
```
Comment (rofe) [inline src/extension/utils.js:379]: I'm sure there's a more elegant way to do this, but here we scan the URL cache for matches with the current owner/repo combo and add them to the array of matches, avoiding duplicates.

**Review by kptdobe — APPROVED** [review]

**Pull request merged**